ProductsManegements/InventoryRepository.cs

```
@@ -0,0 +1,126 @@
+﻿using Dapper;
+using SimpleInventoryManagementSystem.DataAccess;
+using System.Data.SqlClient;
+
+namespace SimpleInventoryManagementSystem.ProductsManagement
+{
+    public class InventoryRepository
+    {
+        private readonly IDbConnectionProvider _connectionProvider;
+        public InventoryRepository(IDbConnectionProvider connectionProvider)
+        {
+            _connectionProvider = connectionProvider ?? throw new ArgumentNullException(nameof(connectionProvider));
+        }
+        public bool AddProduct(Product product)
+        {
+            if (!IsExistProduct(product.Name))
+            {
+                using SqlConnection connection = OpenConnection();
+                string insertQuery = @"
+                                INSERT INTO Product 
+                                    (Name, Quantity, Price, Currency) 
+                                VALUES 
+                                    (@Name, @Quantity, @Price, @Currency)";
+                int rowsAffected = connection.Execute(insertQuery, product);
+                connection.Close();
+                return rowsAffected > 0;
+            }
+            return false;
```

since you know at this point that this false because the product is existed, it's better to return a specific exception or Result.Fail in case your using FluentResult or such library.